Add troubleshooting documentation (#110)
canonical/httprequest-lego-provider: charm/docs/how-to/troubleshoot-api-timeouts.md

Reviewer: I think you need to add this into the `Contents` table in `docs/index.md`

Author: The post doesn't exist yet. Should I create one manually first?

# How to troubleshoot API timeouts

Some of the API calls will execute requests over the network from behind the scenes.

In the event you have a slow network connection this might result in timeouts manifesting as HTTP 500 errors for the API caller. This is more likely for larger Git repositories. If this is your case, adjusting the [`webserver-timeout`](https://charmhub.io/httprequest-lego-provider/configurations#webserver-timeout) configuration can help you solve this.

> You should be able to find a log entry containing the  following for the scenario above:

Reviewer: Do we need to tell the user where they can find the log entry, or how they access it?

Author: Logs should be collected using the COS stack, just like we do for all other charms (PaaS or not). IDK how we are documenting that at the moment but it feels out of scope here to me

# How to troubleshoot API timeouts

> Some of the API calls will execute behind the scenes requests over the network.

Reviewer: ```suggestion
Some of the API calls will execute requests over the network from behind the scenes.
```

> In the event you have a slow network connection this might result in timeouts manifesting as HTTP 500 errors for the API caller. This is more likely the bigger your git repository is. If this is your case, adjusting the [webserver-timeout](https://charmhub.io/httprequest-lego-provider/configurations#webserver-timeout) configuration can help you solve this.

Reviewer: ```suggestion
In the event you have a slow network connection this might result in timeouts manifesting as HTTP 500 errors for the API caller. This is more likely for larger Git repositories. If this is your case, adjusting the [`webserver-timeout`](https://charmhub.io/httprequest-lego-provider/configurations#webserver-timeout) configuration can help you solve this.
```

> Note that if the HTTP Request LEGO provider is sitting behind a reverse proxy, the timeout might be occurring here. In the case of [Nginx ingress integrator](https://charmhub.io/nginx-ingress-integrator), you can change the [proxy-read-timeout](https://charmhub.io/nginx-ingress-integrator/configurations#proxy-read-timeout) configuration to adjust the timeout.

Reviewer: ```suggestion
Note that if the HTTP Request LEGO provider is sitting behind a reverse proxy, the timeout might be occurring here. In the case of [Nginx ingress integrator](https://charmhub.io/nginx-ingress-integrator), you can change the [`proxy-read-timeout`](https://charmhub.io/nginx-ingress-integrator/configurations#proxy-read-timeout) configuration to adjust the timeout.
```
